### ISSUE TITLE
NAB benchmarking train/test split

### ISSUE THREAD

**Issue by jessb126** [sintel-dev/Orion#260]
Hi, I have a question on train/test split of NAB datasets tested in TadGAN paper.

As most of the anomalies in the folders spread across the begin, middle and end of the datasets, how did the team conduct the train/test split? Is it trained/tested on the full set of the data?

If it is trained and tested on the full set of the data, how do you make sure the model is not overfitting?

If a train/test split ratio is selected, which ratio should I use in the orion pipeline?

Thank you and happy holiday! :)
@sarahmish

**Comment by sarahmish** [comment]
Hi @jessb126! 

Since the currently available pipelines are unsupervised in nature, meaning we do not use _ground truth anomalies_ to train the pipelines, we do not split the dataset into training and testing. The notion of "overfitting" still exists in a sense of trying to produce an expected signal very similar to the original, which makes it possible to miss an anomaly.

To learn more about unsupervised pipelines, I recommend reading the last section in [this blogpost](https://t.co/yIFVM1oRwQ?amp=1) where we explain the general principle of how ML is used to detect anomalies.

I hope this answers your question!
Happy holidays! :)

**Comment by jessb126** [comment]
Hi @sarahmish , thank you for your prompt reply.
I am not very clear about your answer. 

You mentioned "we do not use ground truth anomalies to train the pipelines, we do not split the dataset into training and testing". However, for example, in NAB, ground truth anomalies persist across the dataset, such as "realTweets/Twitter_volume_FB.csv", which one of the anomalies from  "2015-03-14 22:12:53.000000"
to "2015-03-17 16:02:53.000000" occur in first half of the dataset.

Do you mean your team has removed those anomalies in NAB datasets by some initial outlier detection technique and impute them with in-range value, or leaving all the anomalies as is in training? If you leave the anomalies in training, cycle consistency generator will minimize the reconstruction output to mimic the original. That's the part that I am unclear about.

Looking forward to your reply.
Regards

**Comment by sarahmish** [comment]
Yes! We leave the anomalies as is in the training, and there is definitely a chance of overfitting the anomalous sequence with the cycle consistency loss. However, we assume that anomalous sequences are infrequent and scarce (by definition of what is anomalous) and therefore we should not fall into this issue.

You can use the [masking primitive](https://sintel.dev/Orion/user_guides/primitives_pipelines/primitives/intervals_to_mask.html) to remove anomalous sequences from training if this is part of your objective.

I hope this answers your question!

**Comment by jessb126** [comment]
Hi Sarah,
Thank you for your clarification, that makes sense.
